Available SDKs Should Return the Empty List if Xcode is Not Installed

chgibb-apple opened the pull request:
#### dff1c2ab2309573594b8e6301b74ac76b8183de0
<pre>
Available SDKs Should Return the Empty List if Xcode is Not Installed
<a href="https://bugs.webkit.org/show_bug.cgi?id=259514">https://bugs.webkit.org/show_bug.cgi?id=259514</a>
rdar://112870698

Reviewed by Alexey Proskuryakov.

If Xcode is not installed, webkitpy’s PlatformInfo#available_sdks method should return the empty list.

* Tools/Scripts/webkitpy/common/system/platforminfo.py:
(PlatformInfo.available_sdks):

Canonical link: <a href="https://commits.webkit.org/266325@main">https://commits.webkit.org/266325@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a80851f98c56a8114bc96f72c99a1e23c0081f97

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13540 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13854 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14185 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15277 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12879 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13619 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16362 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13932 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15564 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13707 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14351 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11462 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/15981 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/13631 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11639 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12217 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/19263 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12714 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12385 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/15607 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12901 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10785 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12169 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3294 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16497 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12744 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/webkitpy/common/system/platforminfo.py:
```diff
@@ -214,7 +214,11 @@ def available_sdks(self):
+
+        try:
+            output = self._executive.run_command(['xcodebuild', '-showsdks'], return_stderr=False)
+        except:
```
Bot says:
```
ERROR: Tools/Scripts/webkitpy/common/system/platforminfo.py:220:  do not use bare 'except'  [pep8/E722] [5]
```

chgibb-apple replied:
Fixed.

aproskuryakov commented on Tools/Scripts/webkitpy/common/system/platforminfo.py:
```diff
@@ -214,7 +214,11 @@ def available_sdks(self):
+        try:
+            output = self._executive.run_command(['xcodebuild', '-showsdks'], return_stderr=False)
+        except:
+            output = ''
```
I think that it would be more elegant to just return an empty list here, not to fool subsequent code.

chgibb-apple replied:
Done.

chgibb-apple committed:
Available SDKs Should Return the Empty List if Xcode is Not Installed
https://bugs.webkit.org/show_bug.cgi?id=259514
rdar://112870698

Reviewed by Alexey Proskuryakov.

If Xcode is not installed, webkitpy’s PlatformInfo#available_sdks method should return the empty list.

* Tools/Scripts/webkitpy/common/system/platforminfo.py:
(PlatformInfo.available_sdks):

Canonical link: https://commits.webkit.org/266325@main
